server/src/main/java/org/elasticsearch/index/mapper/ScriptParams.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.script.ScriptService;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+public final class ScriptParams {
+
+    private ScriptParams() { }
+
+    public static class CompiledScriptParameter<T> implements ToXContent {
+        final Script script;
+        final T compiledScript;
```

Comment:
It is hard to follow what T is here, and I wonder if by any chance this one can be merged with the MapperScript above, or maybe their responsibilities can be clarified. It seems like we need a way to carry around the compiled script, but we do it as part of two different abstractions. Would there be a way to get the compilation out of there, but still have it happen at the right moment?

Reply:
`T` is in fact the mapper script - javadoc will help here I'm sure